mongodb/mongo-cxx-driver
# CXX-2656 Update tests for QEv2 compatibility

## Description

Resolves CXX-2656. Verified by [this patch](https://spruce.mongodb.com/version/6434576b7742ae6f5a6efddb/tasks).

Resolves current Evergreen task failures when testing against latest replica sets due to the following error:

```
Encountered a Queryable Encryption insert/update payload type that is no longer supported
```

## Legacy CSE Tests

This PR moves legacy CSE tests into a proper `legacy` subdirectory for consistency with other unified spec tests with legacy/unified components. Updated CSE test environment variables accordingly.

## CXX-2656 Test Files

This PR syncs the legacy test files with https://github.com/mongodb/specifications/commit/64deb2837a2355f6002775c49b9b6c50c9dc560f with the exception of the following files:

- count.json: see CXX-1940.
- timeoutMS.json: see CXX-2181.
- create-and-createIndexes.json: see CXX-2513.
- create-and-createIndexes.json: see CXX-2513.
- validatorAndPartialFieldExpression.json: see CXX-2513.

## Minimum libmongoc Version

Bumped to https://github.com/mongodb/mongo-c-driver/commit/8aced03a963e70182b79d883e881f239766b09bf to include dependent changes introduced by https://github.com/mongodb/mongo-c-driver/pull/1228, including use of libmongocrypt 1.8.0-alpha0.

## Snapshot Documentation Examples

This PR also introduces a skip condition for snapshot documentation examples. See CXX-2671 for details.

## Miscellaneous

Improved quality of error messages on unexpected test operation success (when an error was expected instead) by emitting the result of the successful command.
